File: blog12_uniqueOccurrences.js
const uniqueOccurrences = arr => {
  // Create a hash
  let occurrences = {}
  // Iterate through the array
  arr.forEach(num => {
    // Store the key-value pairs
    occurrences[num] ? occurrences[num] ++ : occurrences[num] = 1
  })
    // Check if the values in the hash are unique
    return Object.values(occurrences).length === [...new Set(Object.values(occurrences))].length ? true : false 
}

File: blog12_uniqueOccurrences.rb
def unique_occurrences(arr)
#   Create a hash
    occurrences = {}
#   Iterate through the array
    arr.each do |num|
    #   Store the key-value pairs
        occurrences[num] ? occurrences[num] += 1 : occurrences[num] = 1
    end
#   Check if the values in the hash are unique
    occurrences.values.length == occurrences.values.uniq.length
end
